core/domain/skill_domain.py

```diff
@@ -177,6 +236,22 @@ def __init__(self, explanation, worked_examples):
         self.explanation = explanation
         self.worked_examples = worked_examples
 
+    def validate(self):
+        """Validates various properties of the SkillContents object.
+
+        Raises:
+            ValidationError: One or more attributes of skill contents are
+            invalid.
+        """
+        if not isinstance(self.explanation, basestring):
+            raise utils.ValidationError(
+                'Expected skill explanation to be a string, received %s' %
+                self.explanation)
+        if not isinstance(self.worked_examples, list):
```

Comment:
Should test the contents of the list. Apply as much validation as you can think of.

Also the comment about HTML and html_cleaner.clean() probably applies.

Reply:
Done, tested whether each element is a string and added clean().